packages/server/src/common/features/model/model-submission-handler.ts

```diff
@@ -52,6 +56,10 @@ export class ModelSubmissionHandler {
     @inject(CommandStack)
     protected commandStack: CommandStack;
 
+    @inject(ModelValidator)
+    @optional()
+    protected validator: ModelValidator;
```

Comment:
An optional inject should also be defined as optional type:
```suggestion
    protected validator?: ModelValidator;
```